src/tss2-fapi/ifapi_keystore.c

```diff
@@ -445,7 +445,7 @@ ifapi_keystore_initialize(
     }
 
     /* Create user directory if necessary */
-    r = ifapi_io_check_create_dir(keystore->userdir);
+    r = ifapi_io_check_create_dir(keystore->userdir, W_OK);
```

Comment:
Should we add FAPI #defines for these access specifiers? Something like:
```
#define FAPI_WRITE W_OK
#define FAPI_READ R_OK
```
these are Linux/Unix specific and won't work on Windows.

Reply:
I have changed it and added two more more commits. One to delete invalid null hierarchy keys and an update for doxygen.